hs/WebParsing/TimeTableParser.hs

```diff
@@ -36,38 +39,32 @@ insertDepartment :: MonadIO m => ([T.Text], T.Text) -> ReaderT SqlBackend m ()
 insertDepartment (code, name) = 
     insert_ (Tables.Department code name)
 
--- | List of unwanted words to remove when getting department codes.
-wordsToRemove :: [String]
-wordsToRemove = ["courses",
-                "courses]", 
-                "includes",
-                "and",
-                "-",
-                "[",
-                "]",
-                ""]
+-- | Extracts a list of lists of department names followed by unfiltered department codes.
+splitTagsToLists :: String -> Either ParseError [[String]]
+splitTagsToLists input =
+    let tags = endBy line eol
+        line = sepBy cell (char '[')
```

Comment:
@alexbaluta Okay, I thought about this some more and figured out a nicer way of doing it. Look into the `manyTill` and `oneOf` Parsec functions. You can also chain parsers by doing something like

```
(do
  x <- parser1
  y <- parser2
  return (x, y)
)